src/main/scala/piconot/PiconotAPI.scala

```diff
@@ -0,0 +1,66 @@
+import Console.{RED, RESET}
+
+import picolib._
+import picolib.maze._
+import picolib.semantics._
+import piconot.parser._
+import picolib.display._
+import java.io.File
+
+
+@main
+def main(args: String*): Unit = {
+
+  if (args.length != 2) {
+    println(error(usage))
+    sys.exit()
+  }
+
+  // parse the maze file
+  val mazeFileName = args(0)
+  val maze = Maze(getFileLines(mazeFileName))
+
+  // parse the program file
+  val programFilename = args(1)
+  val program = PiconotParser(getFileContents(programFilename))
+
+  // process the results of parsing
+  program match {
+    // Error handling: syntax errors
+    case e: PiconotParser.NoSuccess => println(error(e.toString))
+
+    // If parsing succeeded, create the bot and run it
+    case PiconotParser.Success(t, _) => {
+      object bot extends Picobot(maze, program.get) with TextDisplay
+      bot.run()
+    }
+  }
+
+}
+
+/** A string that describes how to use the program * */
+def usage = "usage: sbt run <maze-file> <rules-file>"
+
+/** Format an error message */
+def error(message: String): String =
+  s"${RESET}${RED}[Error] ${message}${RESET}"
+
+/** Given a filename, get a list of the lines in the file */
+def getFileLines(filename: String): List[String] =
```

Comment:
Nice job writing this getFileContents function! In my implementation I was stuck on using File.separator along with the relative pathname (and just passing that filepath into the Maze constructor), but this is much cleaner